expression/builtin_arithmetic.go

```diff
@@ -548,6 +551,109 @@ func (s *builtinArithmeticDivideDecimalSig) evalDecimal(row []types.Datum) (*typ
 	return c, false, err
 }
 
+type arithmeticIntDivideFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *arithmeticIntDivideFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	tpA, tpB := args[0].GetType(), args[1].GetType()
+	tcA, tcB := numericContextResultType(tpA), numericContextResultType(tpB)
+	if tcA == types.ClassInt && tcB == types.ClassInt {
+		bf := newBaseBuiltinFuncWithTp(args, ctx, tpInt, tpInt, tpInt)
+		if mysql.HasUnsignedFlag(tpA.Flag) || mysql.HasUnsignedFlag(tpB.Flag) {
+			bf.tp.Flag |= mysql.UnsignedFlag
+		}
+		sig := &builtinArithmeticIntDivideIntSig{baseIntBuiltinFunc{bf}}
+		return sig.setSelf(sig), nil
+	}
+	bf := newBaseBuiltinFuncWithTp(args, ctx, tpInt, tpDecimal, tpDecimal)
+	if mysql.HasUnsignedFlag(tpA.Flag) || mysql.HasUnsignedFlag(tpB.Flag) {
+		bf.tp.Flag |= mysql.UnsignedFlag
+	}
+	sig := &builtinArithmeticIntDivideDecimalSig{baseIntBuiltinFunc{bf}}
+	return sig.setSelf(sig), nil
+}
+
+type builtinArithmeticIntDivideIntSig struct{ baseIntBuiltinFunc }
+type builtinArithmeticIntDivideDecimalSig struct{ baseIntBuiltinFunc }
+
+func (s *builtinArithmeticIntDivideIntSig) evalInt(row []types.Datum) (int64, bool, error) {
+	sc := s.ctx.GetSessionVars().StmtCtx
+	b, isNull, err := s.args[1].EvalInt(row, sc)
+	if isNull || err != nil {
+		return 0, isNull, errors.Trace(err)
+	}
+
+	if b == 0 {
+		return 0, true, nil
+	}
+
+	a, isNull, err := s.args[0].EvalInt(row, sc)
+	if isNull || err != nil {
+		return 0, isNull, errors.Trace(err)
+	}
+
+	var (
+		ret int64
+		val uint64
+	)
+	isLHSUnsigned := mysql.HasUnsignedFlag(s.args[0].GetType().Flag)
+	isRHSUnsigned := mysql.HasUnsignedFlag(s.args[1].GetType().Flag)
+
+	switch {
+	case isLHSUnsigned && isRHSUnsigned:
+		ret = a / b
+	case isLHSUnsigned && !isRHSUnsigned:
+		val, err = types.DivUintWithInt(uint64(a), b)
+		ret = int64(val)
+	case !isLHSUnsigned && isRHSUnsigned:
+		val, err = types.DivIntWithUint(a, uint64(b))
+		ret = int64(val)
+	case !isLHSUnsigned && !isRHSUnsigned:
+		ret, err = types.DivInt64(a, b)
+	}
+
+	if err != nil {
+		return ret, false, errors.Trace(err)
+	}
+
+	return ret, false, nil
+}
+
+func (s *builtinArithmeticIntDivideDecimalSig) evalInt(row []types.Datum) (int64, bool, error) {
+	sc := s.ctx.GetSessionVars().StmtCtx
+	a, isNull, err := s.args[0].EvalDecimal(row, sc)
+	if isNull || err != nil {
+		return 0, isNull, errors.Trace(err)
+	}
+
+	b, isNull, err := s.args[1].EvalDecimal(row, sc)
+	if isNull || err != nil {
+		return 0, isNull, errors.Trace(err)
+	}
+
+	c := &types.MyDecimal{}
+	err = types.DecimalDiv(a, b, c, types.DivFracIncr)
+	if err == types.ErrDivByZero {
+		return 0, true, errors.Trace(err)
+	}
+
+	if err != nil {
+		return 0, false, errors.Trace(err)
+	}
+
+	ret, err := c.ToInt()
+	if err == types.ErrOverflow {
```

Comment:
ditto

Reply:
This part have to keep the same. Because function `ToInt` may raise two type error:  `types.ErrOverflow` and `x, ErrTruncated`. And I only wanna process `types.ErrOverflow`, ignore the other.